logic/src/commonMain/kotlin/com/wire/kalium/logic/data/e2ei/E2EIRepository.kt

```diff
@@ -55,7 +57,13 @@ interface E2EIRepository {
     suspend fun getACMENonce(endpoint: String): Either<CoreFailure, String>
     suspend fun createNewAccount(prevNonce: String, createAccountEndpoint: String): Either<CoreFailure, String>
     suspend fun createNewOrder(prevNonce: String, createOrderEndpoint: String): Either<CoreFailure, Triple<NewAcmeOrder, String, String>>
-    suspend fun createAuthz(prevNonce: String, authzEndpoint: String): Either<CoreFailure, Triple<NewAcmeAuthz, String, String>>
+    suspend fun createAuthorization(prevNonce: String, endpoint: String): Either<CoreFailure, AcmeAuthorization>
+
+    suspend fun getAuthorizations(
+        prevNonce: String,
+        authorizationsEndpoints: List<String>
+    ): Either<CoreFailure, Pair<Map<AuthorizationChallengeType, NewAcmeAuthz>, String>>
```

Comment:
Looking at it's behaviour, I find it hard to understand what it's supposed to do exactly without any context. 
I'd need to dive into E2EI documents, maybe.

Could it be a bit documented to understand exactly what it's supposed to expect from it?

It seems to take a `nonce` and process it through the list of authorization endpoints, feeding the output of one as the input of the next, and then returning the final result of the last endpoint. Is it so?
So the order of this list is very important, I'd assume.

Reply:
On another note, what is the `String` of the resulting `Pair`? Seems to be the final `nonce`.

It would be best to return a `data class` with clear indication of what is what.

Something like this:

```kotlin
@JvmInline
value class Nonce(val value: String)

data class AuthorizationResult(
    val challengeResults: Map<AuthorizationChallengeType, NewAcmeAuthz>,
    val nonce: Nonce
)
```